sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -3266,6 +3268,8 @@ && getStorageApiTriggeringFrequency(bqOptions) != null) {
         checkArgument(
             !getAutoSchemaUpdate(),
             "withAutoSchemaUpdate only supported when using STORAGE_WRITE_API or STORAGE_API_AT_LEAST_ONCE.");
+      } else if (getWriteDisposition() == WriteDisposition.WRITE_TRUNCATE) {
```

Comment:
The Storage API sink just won't truncate the table before writing

Reply:
We should probably log this as an error. If we could make breaking changes, I would fail submission here.

In general, a user specifying a configuration that doesn't actually do the thing they would expect is not good.

Reply:
Done. 

Btw we have many other similar mis-configurations in this class that we log a warning for. Should we switch them all to log an error?

Reply:
Probably yes, but we can handle that separately